### PR TITLE
Add to the concepts pages (excluding `Recording data` and `Applications`)

### PR DESCRIPTION
**Pull request by hjstroud**
This PR aims to add the final touches to three of the NanoVer Concepts pages:

- `NanoVer Architecture`
- `Trajectories and Frames`
- `The NanoVer Protocol`

Much of the information in these pages reflects how the core of NanoVer works (i.e. the protocol itself) which (to my knowledge) hasn't changed and is therefore correct. This PR adds some minor additions, rewordings, restructurings and links to the complementary tutorial notebooks that present the same concepts (where appropriate).

### CODE REVIEW

**Review by rhoslynroebuck — DISMISSED** [review]

File: source/concepts/base-protocol.rst [hunk 1]
```diff
@@ -11,23 +11,23 @@ The NanoVer protocol
...
 --------------------
 
-NanoVer provides three services: the state service, the trajectory service,
-and the command service. While none of these services are strictly
+NanoVer provides three services: the **state service**, the **trajectory service**,
+and the **command service**. While none of these services are strictly
 mandatory, some features expect two or three services to cooperate.
 
-The *state service* maintains a state that is shared between the server
+The **state service** maintains a state that is shared between the server
```
Comment (Ragzouken) [inline source/concepts/base-protocol.rst:18]: In all cases "maintains a state" should probably be something clearer along the lines of "coordinates a shared data store"--just "state" alone is probably too vague.

**Review by Ragzouken — CHANGES_REQUESTED** [review]
Looks good to me besides being more explicit about what "state" means

File: source/concepts/base-protocol.rst (hunk 1 above)
Comment (hjstroud) [reply source/concepts/base-protocol.rst]: A good suggestion, I've updated the file to use your suggested wording for all instances of "maintains a state"

**Review by Ragzouken — APPROVED** [review]
Looks good to me with those changes!

**Pull request merged**